### ISSUE TITLE
Mobs are killing each other!!!! (This is CRITICAL!) (BB #739)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#739]
This issue was migrated from bitbucket.
**Original Reporter:** PadreWoW
**Original Date:** 12.11.2014 09:33:37 GMT+0000
**Original Priority:** critical
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739

<hr>

When player attacks two mobs and then run away, use stealth, GM ON, or any other way to remove aggro - mobs start to kill each other

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.11.2014 10:20:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13591242

<hr>

I noticed this a while ago, which revision are you on so I can start from there

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 12.11.2014 10:42:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13591766

<hr>

I use last rev

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 12.11.2014 11:17:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13592501

<hr>

2435+ (f996257651c6)  - bug is HERE

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.11.2014 12:10:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13594036

<hr>

Thanks. That narrows it down enough. I'll fix this after work

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.11.2014 22:11:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13606784

<hr>

[Core/Units] Fix #739

→ <<cset a524af5c029f>>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.12.2014 13:33:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13991697

<hr>

Not Fixed!!!!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.12.2014 13:33:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13991709

<hr>

Not fixed!!!!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.12.2014 13:34:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13991744

<hr>

![1.jpg](https://bitbucket.org/repo/kMeMB/images/3791195470-1.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 01.12.2014 14:51:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13993610

<hr>

Any idea what's causing this?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.12.2014 16:53:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-13996736

<hr>

no ideas

I know only that the error is in this commit:

2435+ (f996257651c6)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.12.2014 21:06:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14001428

<hr>

I can't roll back to this commit because the are some kind of change in "Creature Addon" in one of commits and old core can not be loaded with last DB

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 02.12.2014 03:33:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14006974

<hr>

Try now! :D I hope

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.12.2014 06:28:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14008646

<hr>

Unfortunately not fixed :(

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.12.2014 08:49:51 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14011036

<hr>

The problem is DIRECTLY in this commit:

f996257651c6

I got rollback my core to dd117abe50a5 and no problem there

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 03.12.2014 09:47:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14040909

<hr>

Problem in Unit.cpp - tested

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 03.12.2014 12:17:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14045468

<hr>

I fixed it!

Problem is here:

```
#!c++

     // search nearby enemy before enter evade mode

     if (HasReactState(REACT_AGGRESSIVE))

     {

-        target = SelectNearestTarget();

+        target = SelectNearestTarget(m_CombatDistance ? m_CombatDistance : ATTACK_DISTANCE);

+

         if (target && !IsOutOfThreatArea(target))

             return target;

     }
```

Wrong string:

target = SelectNearestTarget(m_CombatDistance ? m_CombatDistance : ATTACK_DISTANCE);

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 03.12.2014 14:42:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14048910

<hr>

[Core/Creatures] Really fix #739. Thanks to PadreWoW for figuring this one out!

→ <<cset ed8d88904d86>>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 10:13:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14115703

<hr>

Unfortunately I was wrong

Mobs in DUNGEONS killing each other, even if player is alive :((((((

I think they don't use this part of code because they are in dungeon and they can not enter evade mode

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 10:13:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14115704

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 13:46:30 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14117362

<hr>

Problem with dungeon mobs is NOT in commit f996257651c6

very bad :(

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 14:12:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14117583

<hr>

Can you confirm or problem only with my DB ?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 15:07:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14118030

<hr>

I think the problem is with implementation new Spell_threat table

Here in this commit 135b4930594f

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 06.12.2014 15:43:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14118326

<hr>

Looking now, if anything it's a problem with the code changed in Spell.cpp if it's actually from this change. It could be anything now though.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 06.12.2014 15:51:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14118407

<hr>

Which dungeons? I tested in Ramparts, Karazhan unable to reproduce

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 06.12.2014 17:55:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14119329

<hr>

Reproduce:

aggro 2-3 mobs, take some damage and run from them (or GM ON) - can be reproduced easy in Arcatraz and Botanica

Problem is directly in this commit: 135b4930594f - I'm sure

Problem appears if mobs use spells - I think problem with spell threat

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 07.12.2014 16:17:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14128400

<hr>

I'm shure the problem is here:

```
#!c++
if ((m_spellInfo->AttributesEx  & SPELL_ATTR_EX_NO_THREAT) ||
        (m_spellInfo->AttributesEx3 & SPELL_ATTR_EX3_NO_INITIAL_AGGRO))
        return;

    float threat = 0.0f;
    if (SpellThreatEntry const* threatEntry = sSpellMgr.GetSpellThreatEntry(m_spellInfo->Id))
    {
        if (threatEntry->apPctMod != 0.0f)
            threat += threatEntry->apPctMod * m_caster->GetTotalAttackPowerValue(BASE_ATTACK);

        threat += threatEntry->flatMod;
    }
    else
        threat += m_spellInfo->spellLevel;

    if (threat == 0.0f)
        return;


```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.12.2014 15:32:53 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14148363

<hr>

Confirmed problem seems to be here.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.12.2014 01:11:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14158632

<hr>

This is what I did. I reproduced it in the same spot with the same mobs as
your example and couldn't reproduce after this change. I'll take another
look.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.12.2014 01:17:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14158783

<hr>

Hence why we've reverted changes from that commit regarding combat

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.12.2014 02:12:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14159685

<hr>

Spell.cpp maybe.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.12.2014 02:18:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14159789

<hr>

Have you tested

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 09.12.2014 06:18:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14162632

<hr>

I backout this commit (135b4930594f) and returned old "spell_threat" table - all works fine

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 02.01.2015 00:16:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14571107

<hr>

Should be fixed now in  f6a1d05

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 02.01.2015 00:16:59 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14571109

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 20.01.2015 07:39:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14953496

<hr>

NOT FIXED! LAST REV

![Untitled.jpg](https://bitbucket.org/repo/kMeMB/images/3845358385-Untitled.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 20.01.2015 07:39:42 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14953503

<hr>

not fixed!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 20.01.2015 10:43:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14957561

<hr>

mind controlled ?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 20.01.2015 11:15:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-14958372

<hr>

nope

after killing mobs in corridor, Arcatraz Defender started to attack Arcatraz Warder (Warder has spell "Shoot" - EventAI)

All works fine without this commit (135b4930594f)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 16.04.2015 20:30:50 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-17442482

<hr>

you are right desteny :P

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 16.04.2015 20:48:36 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/739/#comment-17442803

<hr>

[Core/Spells] Fix #739. Also cleanup IsPositiveSpell / IsPositiveEffect

→ <<cset f7faf09eea49>>
